pokedex/src/App.js

```diff
@@ -1,109 +1,107 @@
 import "./App.css";
-import React, { useCallback } from "react";
-import Pokemon from "./components/Pokemon";
-import { useSearchParams } from "react-router-dom";
+import React, { useState, useEffect, createRef } from "react";
+import { useOutletContext, useSearchParams } from "react-router-dom";
+import Pokedex from "./components/Pokedex";
+import Displaying from "./components/Displaying";
 
-const App = () => {
-  const [data, setData] = React.useState({});
-  const [meta, setMeta] = React.useState({});
-  const [links, setLinks] = React.useState({});
+const App = ({ context }) => {
   const [searchParams, setSearchParams] = useSearchParams();
   let name = searchParams.get("name");
   let page = searchParams.get("page");
-  const myInput = React.createRef();
+  const [pokedex, setPokedex] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [meta, setMeta] = useState({});
+  const myInput = createRef();
```

Comment:
As I mentioned in the last review, I don't think you need to use a ref here, it could be a controlled component instead. Here is a good example of each:

[](https://medium.com/fasal-engineering/controlled-and-uncontrolled-components-in-react-js-c3111ee0a864)